### ISSUE TITLE
Set username from HTTP headers?

### ISSUE THREAD

**Issue by ericclemmons** [ericclemmons/back-to-the-fixture#9]
```js
fetch("/api", {
  headers: {
    'X-Fixture-Name': 'me'
  }
})
```

This would let us save the file dynamically, rather than relying on `recorder.user` to be set.

I would need to see if this changes things significantly, since _every_ fixture would need to be loaded and headers would not be a differentiator.

**Comment by ericclemmons** [comment]
Instead, it may be better to use `filename` or `fixturePath` or `hash` to do this.

It needs to be bi-directional:

- When recording, we need to know where to record file file, for ease of use.
- When replaying, `nock` needs to know the difference between `asdf.default.json` and `asdf.someuser.js`, and they need to have different headers or something for `nock` to differentiate (since we need to load all mocks).

**Comment by ericclemmons** [comment]
Solved in #11
